pkg/api/v1/resource/helpers.go

```diff
@@ -91,12 +98,24 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 			containerReqs = applyNonMissing(containerReqs, opts.NonMissingContainerRequests)
 		}
 
+		// Is the sidecar feature gate enabled and the init container marked as a sidecar?
+		if opts.SidecarContainersEnabled && container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
```

Comment:
is there a method `IsSideCarContainer`? I think @gjkim42 added it

Reply:
There is, but it also checks the feature flag.  I was passing the flag status into the options struct, but maybe it makes sense to handle the feature flags internally for both this and in-place pod resizing?